### DISCUSSION TITLE
`django-environ`と`python-dotenv`の違いについて

### DISCUSSION THREAD

**Discussion by s1-haya** [42-pong/42-pong#62] [backend]
### 1. 型変換

| 機能        | `python-dotenv` | `django-environ`                             |
| --------- | --------------- | -------------------------------------------- |
| **型変換**   | 手動              | 自動                                           |
| **変換後の型** | 文字列のみ           | 標準的な型変換をサポート（`bool()`, `int()`, `float()`など） |

### 2. **Django以外のプロジェクトでも環境変数を使用できるかどうか**

| 機能              | `python-dotenv`                          | `django-environ`     |
| --------------- | ---------------------------------------- | -------------------- |
| **使用可能なプロジェクト** | Pythonのプロジェクト全般       | Djangoのみ             |
| **利点**          | Pythonのプロジェクト全般  `.env` ファイルを利用できる | Django設定に特化した環境変数の管理 |
### 3. **デフォルト値の設定**

| 機能                         | `python-dotenv`    | `django-environ`                |
| -------------------------- | ------------------ | ------------------------------- |
| **デフォルト値設定**               | 手動                 | 自動                              |
| **`.env` ファイルに記述した値がない場合** | `NULL`             | `default` パラメータを使ってデフォルト値を設定できる |

### 参考にしたページ

[python-dotenv公式](https://github.com/theskumar/python-dotenv)
[django-environ公式](https://django-environ.readthedocs.io/en/latest/)
[django-environ-vs-python-dotenv](https://www.geeksforgeeks.org/comparing-django-environ-vs-python-dotenv/)

**Comment by s1-haya** [comment]
自分の考えとしてはdjangoプロジェクト以外のpythonのソースファイルで環境変数を使用する場合は`python-dotenv`を使い、djangoプロジェクトのみ環境変数を使用する場合は、環境変数を自動で型変換してくれたり、デフォルトで環境変数を設定してくれるので、`django-environ`を使う方がいいと思ったんですけどいかがでしょうか？

**Reply by habvi** [reply]
分かりやすい表ありがとうございます！確かに `python-dotenv` の汎用性は利点ですね
表の内容と以下の理由などから私も `django-environ` で良いと思います！
- 今回の pong は Django から他のフレームワークに変更することはないので汎用性は特にいらなさそう
- せっかく Django を使っているのでこの機会でもないと使わなさそう

(`python-dotenv` でももちろん大丈夫です)
